contrib/src/main/scala/akka/stream/contrib/Valve.scala

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright (C) 2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.stream.contrib
+
+import akka.stream.stage.{GraphStageLogic, GraphStageWithMaterializedValue, InHandler, OutHandler}
+import akka.stream._
+
+import scala.concurrent.{Future, Promise}
+import scala.util.{Failure, Success, Try}
+
+/**
+  * Pause/ Resume a Flow
+  */
+trait ValveSwitch {
+  /**
+    * Change the state of the valve
+    *
+    * @param mode expected mode to switch on
+    *
+    * @return
+    *         Future[true] if the mode change
+    *         Future[false] if the mode didn't change
+    */
+  def flip(mode: SwitchMode): Future[Boolean]
+}
+
+/**
+  * Valve permit to control a flow (pause/resume) by using a ValveSwitch.
```

Comment:
Please indent scaladoc like javadoc (single space before star here).

Also, I think this entire doc could be condensed into something more clear, for example: "Materializes into a [[ValeSwitch]] which provides a the method `flip` that stops or or restarts the flow of elements passing through the stage. As long as the valve is closed it will backpressure. 

Note that closing the valve could result in one element being buffered inside the stage, and if the stream completes or fails while being closed, that element may be lost."